pkg/virt-launcher/virtwrap/agent-poller/agent_poller_test.go

```diff
@@ -55,6 +59,50 @@ var _ = Describe("Qemu agent poller", func() {
 			Id:            "testguestos",
 		}
 	})
+
+	Context("with libvirt API", func() {
+		var agentStore AsyncAgentStore
+		var ctrl *gomock.Controller
+		var mockConnection *cli.MockConnection
+		var mockDomain *cli.MockVirDomain
+
+		BeforeEach(func() {
+			agentStore = NewAsyncAgentStore()
+			ctrl = gomock.NewController(GinkgoT())
+			mockConnection = cli.NewMockConnection(ctrl)
+			mockDomain = cli.NewMockVirDomain(ctrl)
+			mockConnection.EXPECT().LookupDomainByName(gomock.Any()).Return(mockDomain, nil).AnyTimes()
+		})
+
+		It("should store the retrieved guest info when requested", func() {
+			guestInfo := &libvirt.DomainGuestInfo{
+				Interfaces: []libvirt.DomainGuestInfoInterface{{Name: "net0"}},
+				OS:         &libvirt.DomainGuestInfoOS{Name: "fedora"},
+				Hostname:   "test-host",
+				TimeZone:   &libvirt.DomainGuestInfoTimeZone{Name: "EST"},
+				Users:      []libvirt.DomainGuestInfoUser{{Name: "admin"}},
+			}
+
+			mockDomain.EXPECT().Free()
+			mockDomain.EXPECT().GetGuestInfo(INTERFACES|OSINFO|HOSTNAME|TIMEZONE|USERS, uint32(0)).Return(guestInfo, nil)
+
+			fetchAndStoreGuestInfo(INTERFACES|OSINFO|HOSTNAME|TIMEZONE|USERS, mockConnection, &agentStore, "test-domain")
```

Comment:
Generally it's not a good design to call private functions in unit tests.
But I think it can be here for now, and we can improve the test in the next PR, when the rest of the guest agent commands are removed.